bin/oneoff/update_ubuntu/00-preupdate.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+if [[ $(/usr/bin/id -u) -ne 0 ]]; then
+  echo "Not running as root"
+  exit
+fi
+
+# Prior to update, manually execute
+# https://github.com/code-dot-org/code-dot-org/blob/6ecb8f703e13f653d9659e56e98ae55ef49a892e/cookbooks/cdo-nodejs/recipes/default.rb#L40-L48
+# because it expects to be executed with python 3.6, but the update will
+# uninstall that and install python 3.8.
+NODEJS_DIR="/usr/local/nodejs-source-18.16.0"
+if [ -d "$NODEJS_DIR" ]; then
+  (cd $NODEJS_DIR; make uninstall && rm -rf $NODEJS_DIR)
+fi
+
+# Explicitly mark a few packages as held.
+#
+# The mysql packages are important here; otherwise, the update will replace
+# them with mysql 8, and we'll have to manually uninstall that and reinstall
+# the desired versions.
+#
+# The magick packages are slightly less important; they are no longer
+# supported by Ubuntu 20 and will be uninstalled, but can still be reinstalled.
+#
+# However, changes to either of these sets of packages will require the
+# associated gems to be rebuilt with `gem pristine`, or they will fail to
+# recognize the new libraries. To avoid that, hold both mysql and magick.
+#
+# TODO: what should we do for servers that only install a subset of these?
```

Comment:
I think it's probably fine to leave these lines the same for all servers; holding a package that's not installed should just prevent it from being installed, and we unhold these packages afterwards anyways so there will be no long-term implications. Open to ideas if anyone has a better suggestion, though!